rdsbroker/broker.go

```diff
@@ -413,6 +420,21 @@ func (b *RDSBroker) Update(
 	}
 	tagsByName := awsrds.RDSTagsValues(tags)
 
+	// Is this a replica or not?
+	isReplica := false
+	if _, ok := tagsByName[awsrds.TagReplicaOf]; ok {
+		isReplica = true
+	}
+
+	// Validate given parameters for replica promotion
+	if isReplica && updateParameters.PromoteReplica != true {
+		return brokerapi.UpdateServiceSpec{}, fmt.Errorf("Read-replica instances do not support update parameters other than promote_replica")
```

Comment:
are we sure? ones i'm thinking of: `reboot` and `max_allocated_storage`. this leaves rebooting a console-only activity, and i would guess if we autoscale the upstream db, that change wouldn't propagate to the replica. i'm fine if we want to lock down replicas like this; just a thought.

Reply:
Good point, I'll double check how the AWS API handles this.

Reply:
So this is a thing we could do, although I would want to make `promote_replica` mutually exclusive with with `max_allocated_storage` or `reboot` or any other params that could be applied against a read replica. Ultimately I would prefer to just file another ticket onto rds operational support and move forward, since it's a non-trivial amount of work refactoring parts of rdsbroker.Update & Parameters

Reply:
Filed INFRA-936.